### PR TITLE
add kubectl-creyaml plugin

### PR DESCRIPTION
**Pull request by sahil-lakhwani**
Signed-off-by: sahil-lakhwani <sahilakhwani@gmail.com>

This is a plugin to custom resource example YAML file for a given custom resource definition.

- Followed the best practices
- Verified local installation

### CODE REVIEW

File: plugins/creyaml.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+apiVersion: krew.googlecontainertools.github.com/v1alpha2
+kind: Plugin
+metadata:
+  name: creyaml
+spec:
+  version: v0.0.1
+  homepage: https://github.com/sahil-lakhwani/kubectl-creyaml
+  shortDescription: Generate Custom Resource Example Yaml (CREYaml) from Custom Resource Definition
```
Comment (ahmetb) [inline plugins/creyaml.yaml:8]: this is too long, please shorten to half or so.

File: plugins/creyaml.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  description: |
+    This plugin creates example Custom Reource YAML for specified CustomResourceDefinition.
```
Comment (ahmetb) [inline plugins/creyaml.yaml:10]: wrap around 80 characters.

File: plugins/creyaml.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  platforms:
+  - selector:
+      matchLabels:
+        os: linux
+        arch: amd64
+    uri: https://github.com/sahil-lakhwani/kubectl-creyaml/releases/download/v0.0.1/kubectl-creyaml.tar.gz
+    sha256: 6301618b6eb44b664969b31c14c470a1a90d36ee4ba230b54ea625b56d0ae555
+    files:
+    - from: "*"
+      to: "."
```
Comment (ahmetb) [inline plugins/creyaml.yaml:22]: this is not needed

**Comment by sahil-lakhwani** [comment]
hey @ahmetb thanks for the quick feedback. I have added a commit to address your comments.

**Review by corneliusweig — COMMENTED** [review]
The manifest looks good to me, but I think you forgot to include auth providers in your plugin code. This will make the plugin useless for many users, so you should fix it---and that's easy. You just need to add a blank include for the auth package, such as here: https://github.com/corneliusweig/ketall/blob/8df8d676c373ca570b9782ec5117c6e71c96c997/main.go#L24

Ping me once you have released a new version, and I'll merge.

**Comment by sahil-lakhwani** [comment]
Thanks @corneliusweig! I have released a new version with auth providers included and modified the manifest accordingly.

File: plugins/creyaml.yaml [hunk 4]
```diff
@@ -0,0 +1,21 @@
+apiVersion: krew.googlecontainertools.github.com/v1alpha2
+kind: Plugin
+metadata:
+  name: creyaml
+spec:
+  version: v0.0.1
+  homepage: https://github.com/sahil-lakhwani/kubectl-creyaml
+  shortDescription: Generate Custom Resource Example Yaml
```
Comment (ahmetb) [inline plugins/creyaml.yaml:8]: this doesn't need to be all-capitalized.

> Generate custom resource YAML manifest

is ok

File: plugins/creyaml.yaml [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  description: |
+    This plugin creates example Custom Reource YAML for specified
```
Comment (ahmetb) [inline plugins/creyaml.yaml:10]: typo Reource

File: plugins/creyaml.yaml (hunk 4 above)
Comment (sahil-lakhwani) [reply plugins/creyaml.yaml]: changed.

File: plugins/creyaml.yaml (hunk 5 above)
Comment (sahil-lakhwani) [reply plugins/creyaml.yaml]: fixed

**Comment by corneliusweig** [comment]
Ok, looks good. Thanks for contributing.

To reduce your maintenance burden with keeping the manifest updated, please consider automating version bumps with [krew-release-bot](https://krew.sigs.k8s.io/docs/developer-guide/release/automating-updates/).

/lgtm
/approve

**Pull request merged**